src/axom/mir/InterfaceReconstructor.cpp

```diff
@@ -0,0 +1,715 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "InterfaceReconstructor.hpp"
+
+namespace axom
+{
+namespace mir
+{
+
+//--------------------------------------------------------------------------------
+
+InterfaceReconstructor::InterfaceReconstructor()
+{
+  
+}
+
+//--------------------------------------------------------------------------------
+
+InterfaceReconstructor::~InterfaceReconstructor()
+{
+
+}
+
+//--------------------------------------------------------------------------------
+
+mir::MIRMesh InterfaceReconstructor::computeReconstructedInterface(mir::MIRMesh& inputMesh)
```

Comment:
The resulting mesh should probably be passed in as an OUT parameter rather than returned.